src/Analyzer.JsonRuleEngine.UnitTests/AnyOfExpressionTests.cs

```diff
@@ -88,6 +89,72 @@ public void Evaluate_TwoLeafExpressions_ExpectedResultIsReturned(bool evaluation
             Assert.AreEqual(expectedFalse, anyOfEvaluation.EvaluationsEvaluatedFalse.Count());
         }
 
+        [TestMethod]
+        public void Evaluate_SubResourceScopeNotFound_ExpectedResultIsReturned()
```

Comment:
This is also mostly repeated in the tests, right? It probably goes out of the scope of this PR but maybe is worth creating a base class for operators like Any and All, because I'm just realizing those two classes have basically the same code and apparently even some overlapping tests. Should we add an issue for that?

Reply:
Reducing test code would be nice, but since AllOf and AnyOf only have about 20 lines of significant code, it's hard to imagine how that can be meaningfully reduced further without much effort.  I feel like combining those further by introducing an additional base class wouldn't provide any benefit long term.  What benefit do you see?

Reply:
I think we would remove more than 20 lines of repeated code since that test alone is 60 and it's probably not the only one overlapping. Besides that, I think it would make the code more clear

Reply:
I agree.  Combining tests would be very nice.

Combining product code would be nice too, but since there's less than 20 lines of code for the logic of the entire class (for All/AnyOf), I don't know if we would get a good return on the investment in this case.  I could be overestimating the work.  Feel free to create an issue and we can triage it.

Reply:
Created an issue: https://github.com/Azure/template-analyzer/issues/137